modules/subchannel/src/problems/SubChannel1PhaseProblem.C

```diff
@@ -1693,7 +1695,9 @@ SubChannel1PhaseProblem::computeWijResidual(int iblock)
       Vec sol_holder_W;
       LibmeshPetscCall(createPetscVector(sol_holder_W, _block_size * _n_gaps));
       LibmeshPetscCall(populateVectorFromHandle<SolutionHandle>(
-          _prodp, *_P_soln, iblock * _block_size, (iblock + 1) * _block_size - 1, _n_channels));
+          _prodp, *_P_soln, first_node - 1, last_node - 1, _n_channels));
+      LibmeshPetscCall(populateVectorFromDense<libMesh::DenseMatrix<Real>>(
+          _Wij_vec, _Wij, first_node, last_node, _n_gaps));
```

Comment:
this was a miss. 

Do we have a test case for segregated + implicit + cross flows? We should have seen a diff there